client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx

```diff
@@ -142,6 +166,107 @@ class TransferDomainPrecheck extends React.PureComponent {
 		);
 	}
 
+	getRegistrationPeriodMessage() {
+		const { translate, domain } = this.props;
+		const {
+			creationDate,
+			expiryDateOk,
+			inInitialRegistrationPeriod,
+			loading,
+			termMaximumInYears,
+			transferEligibleDate,
+		} = this.state;
+
+		const dateFormat = translate( 'MMM D, YYYY', {
+			comment:
+				'Short month, day and year, like "Dec 5, 2017". This is a moment.js formatted string, ' +
+				'see http://momentjs.com/docs/#/displaying/format/',
+		} );
+
+		let heading = translate(
+			'{{strong}}%(domain)s{{/strong}} can be transferred in %(transferDelayInDays)s days.',
+			{
+				args: {
+					domain: domain,
+					transferDelayInDays: moment( transferEligibleDate ).diff( moment(), 'days' ),
+				},
+				components: {
+					strong: <strong />,
+				},
+			}
+		);
+
+		let message = translate(
+			"You don't have to wait though. Connect your domain to your site now, without transferring it. " +
+				'{{a}}Learn how{{/a}}.',
+			{
+				components: {
+					a: <a href={ support.MAP_EXISTING_DOMAIN } rel="noopener noreferrer" target="_blank" />,
+				},
+			}
+		);
+
+		let reason = null;
+
+		if ( false === expiryDateOk ) {
+			reason = translate(
+				'Transferring this domain would extend the registration period beyond the maximum allowed term ' +
+					'of %(termMaximumInYears)d years. It can be transferred starting %(transferEligibleDate)s.',
```

Comment:
FWIW, for example, for `.fr` this is only 1 year. Which leads to a weird sentence of `Transferring this domain would extend the registration period beyond the maximum allowed term of 1 years` (try transferring-in `php.fr`).